### PR TITLE
Refactor Cross Chain Logic

### LINKED ISSUE #1279: P1.4: THEA Feedback
## 📝 Feedback Description

There are some feedbacks for new THEA UI - 

1. For a selected source chain and asset, destination chain is not being displayed properly. It always shows all the chains no matter what is source chain or asset. ✅

Image

2. Balance on Asset PopUp & Input field is different. ✅

 

3. `Balance` and `Available` looks misleading. Both shows different values as well. ✅

4. Sender Wallet & Source wallet both are same things. It should be Source and Destination wallet instead. ✅

5.  Link for transfer history page

6. Upon selecting GLMR/EVM network deposits, the substrate account has to be deselected and a selection component for ETH wallet to be automatically triggered

7. Notification message to be updated: Deposit Success. Processed transactions take a few minutes to appear in History" ✅

8. Refactor THEA provider ✅

9. Amount formatting issue ✅

10. Fix Amount in Cross chain History table ✅

## 🌍 Your Environment

| Software         | Version(s)                  |
| :----------------: | :---------------------------: |
| Project/Package  |           Orderbook Frontend                  |
| Browser          | [x] Chrome [] Firefox [] Safari |
| Operating System | [] Windows [x] MacOS [] Linux   |

### LINKED ISSUE #1289: Couple minor things on internal page
"Fund account" area still points to old thea.polkadex.trade rather then new bridge page.

image

Transaction section for the bridge is titled "Thea History", but on the bridge page there is no mention of the word Thea. Users may not know these are related.
image

image

Also is it Thea or THEA. Should try to be consistent for branding purposes.

### PR DESCRIPTION
**Pull request by ap211unitech**
## Description

There are some feedbacks for new THEA UI - 

1. For a selected source chain and asset, destination chain is not being displayed properly. It always shows all the chains no matter what is source chain or asset. ✅

Image

2. Balance on Asset PopUp & Input field is different. ✅

 

3. `Balance` and `Available` looks misleading. Both shows different values as well. ✅

4. Sender Wallet & Source wallet both are same things. It should be Source and Destination wallet instead. ✅

5.  Link for transfer history page

6. Upon selecting GLMR/EVM network deposits, the substrate account has to be deselected and a selection component for ETH wallet to be automatically triggered

7. Notification message to be updated: Deposit Success. Processed transactions take a few minutes to appear in History" ✅

8. Refactor THEA provider ✅

9. Amount formatting issue ✅

10. Fix Amount in Cross chain History table ✅

https://github.com/Polkadex-Substrate/Polkadex-Orderbook-Frontend/assets/65214523/9e2e7826-a925-47c1-a5af-a9fec841257c

## Checklist

- [x] Included link to corresponding [Polkadex Orderbook Frontend Issue](https://github.com/Polkadex-Substrate/Polkadex-Orderbook-Frontend/issues).
- [x] I have tested these changes thoroughly.
- [x] I have requested a review from at least one other contributor.

### CODE REVIEW

**Review by minikas — CHANGES_REQUESTED** [review]
- There is a problem with the validations.
- The max button is no longer visible

https://github.com/Polkadex-Substrate/Polkadex-Orderbook-Frontend/assets/12574469/550c460d-f959-4556-8e0e-4b164d998da5

**Comment by ap211unitech** [comment]
> * There is a problem with the validations.
> * The max button is no longer visible
> 
>  Screen.Recording.2024-05-22.at.07.57.03.mov

@minikas This is fixed now. Please review again.

File: apps/hestia/src/components/thea/Help/index.tsx [hunk 1]
```diff
@@ -2,23 +2,42 @@
...
-      className="flex items-center max-md:flex-col border-t border-primary"
+      className="flex items-start max-md:flex-col max-md:gap-4 max-w-[900px] flex-1 mx-auto w-full mt-20 max-md:pb-4"
     >
-      <Card title="Having Trouble?" description="Feel free to get in touch.">
-        <Link href="https://discord.gg/G4KMw2sGGe" target="_blank">
-          <Button.Icon variant="outline">
-            <RiExternalLinkLine className="w-full h-full" />
-          </Button.Icon>
-        </Link>
+      <Card
```
Comment (minikas) [inline apps/hestia/src/components/thea/Help/index.tsx:15]: Make the entire component clickable

**Review by minikas — CHANGES_REQUESTED** [review]

File: apps/hestia/src/components/thea/Help/index.tsx [hunk 2]
```diff
@@ -2,23 +2,42 @@
...
+      >
+        <Button.Icon
+          size="2sm"
+          appearance="secondary"
+          className="rounded-md bg-secondary-base pointer-events-none"
+        >
+          <RiBookOpenLine className="w-full h-full" />
+        </Button.Icon>
+      </Card>
+      <Card
```
Comment (minikas) [inline apps/hestia/src/components/thea/Help/index.tsx:28]: Same here

**Review by minikas — CHANGES_REQUESTED** [review]

**Review by minikas — CHANGES_REQUESTED** [review]
Check the alignment
Screenshot 2024-05-23 at 08 13 17

File: apps/hestia/src/components/thea/Help/index.tsx (hunk 1 above)
Comment (minikas) [reply apps/hestia/src/components/thea/Help/index.tsx:15]: Increase the size of the title

**Review by minikas — CHANGES_REQUESTED** [review]

File: apps/hestia/src/components/thea/Help/index.tsx (hunk 1 above)
Comment (ap211unitech) [reply apps/hestia/src/components/thea/Help/index.tsx:15]: Done

File: apps/hestia/src/components/thea/Help/index.tsx (hunk 2 above)
Comment (ap211unitech) [reply apps/hestia/src/components/thea/Help/index.tsx:28]: Done

**Comment by ap211unitech** [comment]
> Check the alignment Screenshot 2024-05-23 at 08 13 17

@minikas This is fixed now.

**Review by minikas — APPROVED** [review]
LGTM

**Review by nuel77 — APPROVED** [review]

**Pull request merged**